src/scripts/main.js

```diff
@@ -1 +1,69 @@
 'use strict';
+
+document.oncontextmenu = () => false;
+
+const buttonState = { left: false, right: false };
+let resetTimeout;
+
+const promise1 = new Promise((resolve, reject) => {
+  document.addEventListener('mousedown', (e) => {
+    if (e.buttons === 1) {
+      return resolve('First');
+    }
+  });
+
+  setTimeout(() => reject(new Error('First promise was rejected')), 3000);
+});
+
+const promise2 = new Promise((resolve) => {
+  document.addEventListener('mousedown', (e) => {
+    if (e.buttons === 1 || e.buttons === 2) {
```

Comment:
The condition 'e.buttons === 1 || e.buttons === 2' in promise2 will resolve the promise if either the left or right mouse button is pressed. This might not be the intended behavior if you want to distinguish between the two buttons.